### DISCUSSION TITLE
Mining

### DISCUSSION THREAD

**Discussion by lstebner** [jeremyckahn/farmhand#105] [Ideas]
A place to discuss how mining could work in Farmhand

**Comment by lstebner** [comment]
@jeremyckahn mentioned possibly using the field that already exists, so an idea that comes to mind with that is to spawn rocks on the outside edges of the field. tapping them could mine them, and what comes out would be based on a random spawn chance. 

another idea that comes to mind is to have an area similar to the field in which rocks spawn. mining them could work with a similar mechanic. 

i imagine with either we could have rocks of different types that drop different items, and maybe take different amounts of effort to crack.

**Comment by jeremyckahn** [comment]
I like the idea of spawning rocks and tapping to mine them! I'm not sure if the side of the field is the place for them, as larger fields on smaller screens are already a bit tight.

Your idea got me thinking about how mining might be more closely integrated with the farming mechanics. Similarly to how crops are planted, what if there was a similar mechanic for ores? Off the top of my head, I could see a "Summoning Pebbles" item that you plant similarly to a seed, but requires no water and results in a randomly-chosen ore after a few days. We could even have tiers of Summoning Pebbles that result in different tiers of resulting ores.

I think what I've suggested could use some refinement (🙃), but I'd like to explore the path of integrating mining into the field a little further.

-------------------------------

It's worth mentioning that what I'm suggesting could make the field a bit cramped, but this could be mitigated by additional field expansion (#86).

**Reply by lstebner** [reply]
interesting idea. how does that fit with the "world" of Farmhand? sounds a bit magical, is that a direction you want to take? 

you made me think of one different option at least though, introducing a new tool (shovel?) that can be used on empty farm tiles to potentially dig up a rock, which can be cracked open for some sort of material.

**Comment by jeremyckahn** [comment]
>how does that fit with the "world" of Farmhand? sounds a bit magical, is that a direction you want to take?

With Rainbow Cows/Milk/Fertilizer, Farmhand is already going in a pretty "magical" direction and I quite like that. 🙂 I'd like to go further down this path, as I think fanciful design aspects will make this game more fun that if we tried to keep it more of a simulator. I also like how the game becomes more fanciful as it progresses; I think it serves as a nice surprise for players. I think we should be open to Farmhand becoming very disconnected from reality (in a good way) for the late/meta game. 🪄

>you made me think of one different option at least though, introducing a new tool (shovel?) that can be used on empty farm tiles to potentially dig up a rock, which can be cracked open for some sort of material.

This is really interesting! Maybe instead of buying Summoning Pebbles from the shop, ores could be dug up from empty plots? I'm thinking that an empty plot could be excavated once per day, and an excavated plot may or may not yield an ore, rock, or some other treasure. An excavated plot would be unusable for anything (including planting crops) until the next day once it resets.

**Reply by lstebner** [reply]
awesome, totally agree with regards to keeping things interesting rather than sim-like. will keep this in mind going forward!

question, just for clarification for me; a "plot" is a single tile in the field, correct? if so, i think that's a good way to go for v1 of mining!

**Reply by jeremyckahn** [reply]
> a "plot" is a single tile in the field, correct?

That’s correct! A plot can contain a variety of different types on content, like a crop, a sprinkler, a scarecrow… or an ore! 🪨

**Reply by lstebner** [reply]
for anyone else reading this discussion, @jeremyckahn and I chatted a bit on discord and decided a few things.

1. going with a new tool, the "Shovel", to allow digging in plots will be the mechanic (there are many details to work out here still, but it's a direction)
2. the shovel will be acquired as a reward to the player for repaying their initial loan to the bank
3. "get diggiong!" or something can be an achievement for the player acquiring the shovel. this would also add a little teaser into the game that the shovel is coming eventually

**Comment by jeremyckahn** [comment]
There is now a Mining Project to track the work for this feature: https://github.com/jeremyckahn/farmhand/projects/5

**Reply by burning-eggs** [reply]
Is this open for development?

**Reply by jeremyckahn** [reply]
Yes! @lstebner is actively working on #131 (see #146). I have made "shoveled plot" art in #153. Currently, #132, #134, and #135 could at least be started on. @lstebner has been leading the development of the mining feature, so I will leave it up to him if he'd like some support with these open tickets or if he'd prefer to own the remainder of the work here. In case of the latter, there's a number of other fun tickets I could help you find if you're interested, @flying-dojo!

**Reply by lstebner** [reply]
@flying-dojo you hinted that you were also good with art, so #135 seems like a good one to work on to me since it involves adding some items to the game, and they each need a visual as well. i think there may be some details we need to work out there with item values, and i do not know how comfortable you are in javascript, but one of us can help you find your way around if needed. if you decide to work on just assign it to yourself please so we know. thanks!

**Comment by lstebner** [comment]
i just started to work on #135 and it feels a little bit lackluster to stop at just having ore. it's definitely some scope creep, but i think we should add in "ingots", or more simply "bars" of materials to the workshop. we could make this an unlock for selling lots of ores so that it isn't all available at the same time. maybe we can think of a few other workshop items those can also be used for.

i can't help but mention a "smelter" as well for the production of these ingots, and possibly other things in the future (maybe some other field tools even?). i imagine once the shovel is unlocked, the smelter could become available for sale in the shop, and having it + N number of ore sold would be the unlock req for the ingots. the smelter could just be something that is bought and unlocks the path to more recipes, i don't think it needs any sort of new screen or interface.

**Reply by jeremyckahn** [reply]
I am on board with this! It does blow out the scope a bit, but in this case I think it's worth having a feature we feel is a reasonably complete addition to the game. Transforming raw ore materials into ingredient materials like bars is an established farming game mechanic that would fit well here. Before moving ahead, I just want to get a sense of how this will impact the overall game.

I thing @fgggg-a raised some good points about upgrading tools. It's another idea that's been established in other games, and I think we should use it in Farmhand as well. Upgrading the scythe in particular could be interesting to produce higher crop yields. In any case, as the mining feature comes together, I can see it opening the door to all manner of item upgrades. I don't think we need to cover everything before shipping the first iteration of this feature, but I'd be on board with targeting one upgradable item being the end goal for V1 of mining. @lstebner, what do you think about this being the player goal for V1 of mining?

>i can't help but mention a "smelter" as well for the production of these ingots, and possibly other things in the future (maybe some other field tools even?). i imagine once the shovel is unlocked, the smelter could become available for sale in the shop, and having it + N number of ore sold would be the unlock req for the ingots. the smelter could just be something that is bought and unlocks the path to more recipes, i don't think it needs any sort of new screen or interface.

+1 to this! 👍

**Reply by lstebner** [reply]
scythe upgrade sounds like a good plan to me. if we introduce iron, steel, and gold, we could use each as a tier of upgrade, and each could slightly increase yield. though, since most of our yields are 1 item, increasing a few % isn't going to do anything, so we have to consider 2x, 3x, etc i think. that could actually be an _alternative_ to #86 . instead of more fields, we could just embrace having more yield per plot.

**Reply by jeremyckahn** [reply]
I agree with all of this! Upgrading field tools to produce 2/3/4/+ X yields sounds like a great mechanic that fits well with what we have and can be expanded on, both conceptually and technically, quite a bit. Let's do it! 👍

**Comment by fgggg-a** [comment]
I think making tabs Is  a good idea
Since we're gonna need smelter and a forge

**Comment by fgggg-a** [comment]
1 new area
Mining
2 new sub sections
Smelter and forge
For craft

You use powdered coal and not coal
Stone is useless
2 Powdered coal and 20 powdered iron in forge make bucketswhich are used to milkcow
With 10 iron bar And 1 silver rod You can make a steel pickaxe which makes yeild 2.5* better
You can buy a normal pickake for 1000$
With 1 powdered coal 1 diamond and 1 powdered gold in forge you 
Can make a ring

**Reply by jeremyckahn** [reply]
I think upgrading items is a great idea! See: https://github.com/jeremyckahn/farmhand/discussions/105#discussioncomment-1013323

**Reply by lstebner** [reply]
the "powdered" idea is interesting. as is the ring. rings i think could be a different discussion entirely actually, that could go a lot of ways.

**Reply by jeremyckahn** [reply]
+1! I think mining opens the door to a whole realm of upgrade mechanics for the game. And we can add layers of mechanics over time, too. We shouldn't feel pressured to get it all in the initial implementation! 🙂

**Comment by lstebner** [comment]
hey @jeremyckahn we mentioned coal and filed an issue for it, but we never talked about where it comes from. is it just another item that can be dug up?

**Reply by jeremyckahn** [reply]
Oh yeah, let's stick the [Stardew Valley model of digging it up as an item](https://stardewvalleywiki.com/Coal#:~:text=Coal%20can%20sometimes%20be%20found%20by%20mining%20rocks). Once #116 is shipped we can also add [crafting coal from wood](https://forager.fandom.com/wiki/Materials#Wood)!

**Comment by Moira2** [comment]
您的邮件已收到，谢谢
